### PR TITLE
Multinode support for osp

### PR DESCRIPTION
**Pull request by rmohr**
@cynepco3hahue if you like you can pick that up to add multinode support for our osp cluster. The node needs to be started with --reverse so that the nodes are accessible before the master is up. Then the master discovers all nodes via dns and adds them to its inventory. Finally the master provisiones the nodes. It takes about two minutes to provision a node since all dependencies are already there.

### CODE REVIEW

File: os-3.9-masteronly/run.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+set -e
+
+../cli/cli run --nodes 2 --background --memory 4096M --base kubevirtci/os-3.9
```
Comment (rmohr) [inline os-3.9-masteronly/run.sh]: Here --reverse is needed.

File: os-3.9/run.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+set -e
+
+../cli/cli run --nodes 2 --background --memory 4096M --base kubevirtci/os-3.9-multinode
```
Comment (rmohr) [inline os-3.9/run.sh]: Here --reverse is needed.

File: os-3.9/scripts/node01.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  node=$(printf "node%02d" ${i})
+  num=$(printf "%02d" ${i})
+  set +e
+  ping -4 ${node} -c 1 
+  set -e
+  if [ $? -ne 0 ]; then
+      break
+  fi
+  echo "Found ${node}. Adding it to the inventory."
+  echo "${node} openshift_node_labels=\"{'region': 'infra','zone': 'default'}\" openshift_schedulable=true openshift_ip=192.168.66.1${num}"
```
Comment (rmohr) [inline os-3.9/scripts/node01.sh]: `>> $inventory_file` is missing to add it to the inventory.

**Comment by cynepco3hahue** [comment]
@rmohr Hm do we need to clean all master stuff before it? Because we reprovision image as master, I saw you stop and disable master services, does it enough?

**Comment by rmohr** [comment]
Yes, tried it by hand a few times. No cleaning necessary.

File: os-3.9/scripts/node01.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+# Update inventory
+echo "[new_nodes]" >> ~/inventory
+sed -i '/[OSEv3:children]/a new_nodes' ~/inventory
+
+for i in "$(seq 2 100)"; do
+  node=$(printf "node%02d" ${i})
+  num=$(printf "%02d" ${i})
+  set +e
+  ping -4 ${node} -c 1 
+  set -e
```
Comment (rmohr) [inline os-3.9/scripts/node01.sh]: That overrides $?

**Comment by cynepco3hahue** [comment]
I think we must just delete `masteronly` directory, because the case with NUM_NODE=0, will cover it, all other stuff looks good.

File: os-3.9/scripts/node01.sh [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+# Update DHCP lease, will also update DNS servers
+dhclient
+
+openshift_ansible_dir="/root/openshift-ansible"
+inventory_file="/root/inventory"
+
+# Update inventory
+echo "[new_nodes]" >> $inventory_file
+sed -i '/[OSEv3:children]/a new_nodes' $inventory_file
```
Comment (rmohr) [inline os-3.9/scripts/node01.sh]: this seems to not work as expected

File: os-3.9/scripts/node01.sh [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+nodes_found="false"
+for i in "$(seq 2 100)"; do
```
Comment (rmohr) [inline os-3.9/scripts/node01.sh]: Still an issue in the for-loop.

File: os-3.9/scripts/node01.sh (hunk 5 above)
Comment (rmohr) [reply os-3.9/scripts/node01.sh]: fixed

File: os-3.9/scripts/node01.sh (hunk 6 above)
Comment (rmohr) [reply os-3.9/scripts/node01.sh]: fixed

**Comment by rmohr** [comment]
@cynepco3hahue verified with `--nodes 3`.

Relevant dns-discover output:
```bash
--- node02 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.210/0.210/0.210/0.000 ms
Found node02. Adding it to the inventory.
PING node03 (192.168.66.103) 56(84) bytes of data.
64 bytes from node03 (192.168.66.103): icmp_seq=1 ttl=64 time=0.449 ms

--- node03 ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.449/0.449/0.449/0.000 ms
Found node03. Adding it to the inventory.
ping: node04: Name or service not known
```

All nodes are ready:
```bash
cli ssh node01
2018/03/14 09:25:32 Waiting for host: 192.168.66.101:22
2018/03/14 09:25:32 Connected to tcp://192.168.66.101:22
[vagrant@node01 ~]$ sudo /usr/local/bin/oc get nodes
NAME      STATUS    ROLES     AGE       VERSION
node01    Ready     master    11m       v1.9.1+a0ce1bc657
node02    Ready     <none>    37s       v1.9.1+a0ce1bc657
node03    Ready     <none>    37s       v1.9.1+a0ce1bc657
```

**Comment by cynepco3hahue** [comment]
@rmohr Cool, then I will merge it and update relevant images.

**Pull request merged**